### PR TITLE
Enable ARM Support in Presto

### PR DESCRIPTION
**Pull request by ChunxuTang**
Support ARM (e.g. Apple Silicon chips) in Presto.

Co-authored-by: Jun Luo <sea11509@mail.ustc.edu.cn>
Co-authored-by: striverpan <616358337@qq.com>
Co-authored-by: Beinan Wang <soros.wang@yahoo.com>

Test plan - Have tested in the local environments.

```
== RELEASE NOTES ==

General Changes
* Fix several issues when building Presto on ARM based processors.

```

### CODE REVIEW

**Comment by ChunxuTang** [comment]
Note: Requires https://github.com/prestodb/presto-hadoop-apache2/pull/59 and a new release of `presto-hadoop-apache2`.

**Comment by v-jizhang** [comment]
@prestobot kick off tests

File: pom.xml [hunk 1]
```diff
@@ -1201,6 +1201,12 @@
                 <version>7.0.0.jre8</version>
             </dependency>
 
+            <dependency>
+                <groupId>org.fusesource.jansi</groupId>
```
Comment (yingsu00) [inline pom.xml:1205]: @ChunxuTang Could you enlighten me why we need jansi here?

Comment (ChunxuTang) [reply pom.xml:1205]: Hi @yingsu00, thanks for the comment!
`presto-cli` implicitly depends on jansi. We need to explicitly marks it with a version that supports both X86 and ARM architectures.
Without this change, we'll meet some errors when running the `presto-cli`, e.g. https://github.com/prestodb/presto/issues/17150 and https://github.com/prestodb/presto/issues/17146.

**Comment by hbazan-pp** [comment]
is there an ETA for this PR to be merged and tagged? or is there another approach to make presto work on ARM chips?
Thanks

**Review by yingsu00 — APPROVED** [review]

**Review by yingsu00 — COMMENTED** [review]
@ChunxuTang Sorry for the delay! I asee Jansi is now in 2.4.x, do you want to update and test it again?

**Comment by yingsu00** [comment]
@wanglinsong Llinsong, will you be able to take a look at the cancelled jenkins test? Why was it cancelled?

**Comment by wanglinsong** [comment]
I have disabled Jenkins pipeline for PRs.

**Comment by wanglinsong** [comment]
The PR message is not complete. This just enables the aarch64 support. There will be lots other updates needed to make build and test pass on aarch64.

**Comment by yingsu00** [comment]
> The PR message is not complete. This just enables the aarch64 support. There will be lots other updates needed to make build and test pass on aarch64.

@wanglinsong I can't merge a PR if some tests fail, even if it's seemingly irrelevant. Is it possible to remove it from the test lists then?

**Comment by yingsu00** [comment]
> I have disabled Jenkins pipeline for PRs. It doesn’t block the merge now.

WHat does the Jenkins tests test? Are they the same tests as the others ?

**Comment by wanglinsong** [comment]
> 

Jenkins pipeline build docker images. No testing.

**Comment by yingsu00** [comment]
> > 
> 
> Jenkins pipeline build docker images. No testing.

Should it be ran for this PR? This PR would change the docker image in my understanding so it should be ran. Is this not right?

**Comment by wanglinsong** [comment]
> 

https://ci.ahana.dev/blue/organizations/jenkins/prestodb%2Fpresto/detail/PR-18960/5/pipeline/

**Comment by majetideepak** [comment]
@imjalpreet I see a commit for arm64 here https://github.com/prestodb/presto-hadoop-apache2/commit/70d1090c43ccca015647a072adf046576eb2da38
Can you share particular query that will throw the above error?

**Comment by imjalpreet** [comment]
@majetideepak Yes, that commit is required to complete the support of Presto on ARM. But it is not part of the presto-hadoop-apache2 version Presto is currently using. That's why we need a new release of presto-hadoop-apache2 so that we can use that in Presto.

**Comment by imjalpreet** [comment]
> Can you share a particular query that will throw the above error?

There would be an issue while starting the server since it would not be able to find the native libs for arm64

**Comment by majetideepak** [comment]
@imjalpreet Looks like this is a recent change. Can you open a new issue with the current version being used?
How are the native libs for x86_64 generated for the current version?

**Comment by majetideepak** [comment]
I see this is a release issue. Thanks for clarifying.

**Comment by Akanksha-kedia** [comment]
i m still facing issue chip : Apple M1 Pro but issue :
java.lang.ExceptionInInitializerError
at io.prestosql.plugin.hive.HdfsEnvironment$$FastClassByGuice$$2c8553d4.newInstance()
at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
at com.google.inject.internal.ConstructorInjector.access$000(ConstructorInjector.java:32)
at com.google.inject.internal.ConstructorInjector$1.call(ConstructorInjector.java:98)
at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:112)
at io.airlift.bootstrap.LifeCycleModule.provision(LifeCycleModule.java:54)
at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:120)
at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:93)
at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
at com.google.inject.internal.ConstructorInjector.access$000(ConstructorInjector.java:32)
at com.google.inject.internal.ConstructorInjector$1.call(ConstructorInjector.java:98)
at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:112)
at io.airlift.bootstrap.LifeCycleModule.provision(LifeCycleModule.java:54)
at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:120)
at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:93)
at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:213)
at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:184)
at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:111)
at com.google.inject.Guice.createInjector(Guice.java:87)
at io.airlift.bootstrap.Bootstrap.initialize(Bootstrap.java:276)
at io.prestosql.plugin.hive.InternalHiveConnectorFactory.createConnector(InternalHiveConnectorFactory.java:117)
at io.prestosql.plugin.hive.InternalHiveConnectorFactory.createConnector(InternalHiveConnectorFactory.java:77)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at io.prestosql.plugin.hive.HiveConnectorFactory.create(HiveConnectorFactory.java:63)
at io.prestosql.connector.ConnectorManager.createConnector(ConnectorManager.java:353)
at io.prestosql.connector.ConnectorManager.createCatalog(ConnectorManager.java:212)
at io.prestosql.connector.ConnectorManager.createCatalog(ConnectorManager.java:204)
at io.prestosql.connector.ConnectorManager.createCatalog(ConnectorManager.java:190)
at io.prestosql.metadata.StaticCatalogStore.loadCatalog(StaticCatalogStore.java:88)
at io.prestosql.metadata.StaticCatalogStore.loadCatalogs(StaticCatalogStore.java:68)
at io.prestosql.server.Server.doStart(Server.java:119)
at io.prestosql.server.Server.lambda$start$0(Server.java:73)
at io.prestosql.$gen.Presto_unknown____20240504_042839_1.run(Unknown Source)
at io.prestosql.server.Server.start(Server.java:73)
at io.prestosql.server.PrestoServer.main(PrestoServer.java:38)
Caused by: java.lang.RuntimeException: failed to load Hadoop native library
at io.prestosql.hadoop.HadoopNative.requireHadoopNative(HadoopNative.java:59)
at io.prestosql.plugin.hive.HdfsEnvironment.(HdfsEnvironment.java:39)
... 52 more
Caused by: java.lang.RuntimeException: library not found: /nativelib/Mac_OS_X-aarch64/libhadoop.dylib
at io.prestosql.hadoop.HadoopNative.loadLibrary(HadoopNative.java:89)
at io.prestosql.hadoop.HadoopNative.requireHadoopNative(HadoopNative.java:47)
... 53 more

Screenshot 2024-05-04 at 10 01 10 AM
also i tried exporting export DYLD_LIBRARY_PATH=/path/

Downloaded the library from the provided link and saved it to a suitable directory, say /Users/ak/lib.

Set the DYLD_LIBRARY_PATH environment variable to include the directory where saved the libhadoop.dylib file:

export DYLD_LIBRARY_PATH=/Users/ak/lib

**Pull request merged**